modules/images/webp-uploads/load.php

```diff
@@ -11,42 +11,92 @@
- * `webp_uploads_valid_image_mime_types`. If the original image is one of the mimes from
- * `webp_uploads_valid_image_mime_types` the image is just added to the `sources` property and  not
- * created again. If the uploaded attachment is not a valid image this function does not alter the
- * metadata of the attachment, on the other hand a `sources` property is added.
+ * `webp_uploads_get_supported_image_mime_transforms`. If the original image is one of the mimes from
+ * `webp_uploads_get_supported_image_mime_transforms` the image is just added to the `sources` property and not
+ * created again. If the uploaded attachment is not a supported mime by this function, the hook does not alter the
+ * metadata of the attachment. In addition to every single size the `sources` property is added at the
+ * top level of the image metadata to store the references for all the mime types for the `full` size image of the
+ * attachment.
  *
  * @since 1.0.0
  *
  * @see   wp_generate_attachment_metadata()
- * @see   webp_uploads_valid_image_mime_types()
+ * @see   webp_uploads_get_supported_image_mime_transforms()
  *
  * @param array $metadata      An array with the metadata from this attachment.
  * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
  * @return array An array with the updated structure for the metadata before is stored in the database.
  */
 function webp_uploads_create_sources_property( array $metadata, $attachment_id ) {
-	// Make sure we have some sizes to work with, otherwise avoid any work.
-	if ( empty( $metadata['sizes'] ) || ! is_array( $metadata['sizes'] ) ) {
-		return $metadata;
-	}
 	// This should take place only on the JPEG image.
 	$valid_mime_transforms = webp_uploads_get_supported_image_mime_transforms();
-
 	// Not a supported mime type to create the sources property.
 	$mime_type = get_post_mime_type( $attachment_id );
-	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) ) {
+	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) || ! is_array( $valid_mime_transforms[ $mime_type ] ) ) {
 		return $metadata;
 	}
 
 	$file = get_attached_file( $attachment_id, true );
-
 	// File does not exist.
 	if ( ! file_exists( $file ) ) {
 		return $metadata;
 	}
 
-	$dirname = pathinfo( $file, PATHINFO_DIRNAME );
+	// Make sure the top level `sources` key is a valid array.
+	if ( ! isset( $metadata['sources'] ) || ! is_array( $metadata['sources'] ) ) {
+		$metadata['sources'] = array();
+	}
+
+	if ( empty( $metadata['sources'][ $mime_type ] ) ) {
+		$metadata['sources'][ $mime_type ] = array(
+			'file'     => wp_basename( $file ),
+			'filesize' => filesize( $file ),
+		);
+		wp_update_attachment_metadata( $attachment_id, $metadata );
+	}
+
+	$original_size_data = array(
+		'width'  => isset( $metadata['width'] ) ? (int) $metadata['width'] : 0,
+		'height' => isset( $metadata['height'] ) ? (int) $metadata['height'] : 0,
```

Comment:
If `$metadata['width']` or `$metadata['height']` are not set, this entire block of code wouldn't really work, as dimensions of 0x0 don't make sense. I think it would make more sense to wrap the entire block of code to generate additional formats for the full-size image into a `isset( $metadata['width'] ) && isset( $metadata['height'] )` conditional instead, so that the rest of the dependant code isn't run if one of these is not set.

Reply:
Great point, let me add a condition to prevent moving forward if both dimensions are not defined.